src/services/sync.service.ts

```diff
@@ -115,18 +115,30 @@ export class SyncService {
     private removeDuplicateUsers(users: UserEntry[]) {
         const uniqueUsers = new Array<UserEntry>();
         const processedUsers = new Map<string, string>();
+        const processedDeletedUsers = new Map<string, string>();
         const duplicateEmails = new Array<string>();
 
         // UserEntrys with the same email are ignored if their properties are the same
-        // UserEntrys with the same email but different properties will throw an error
+        // UserEntrys with the same email but different properties will throw an error 
```

Comment:
Update to describe new logic, e.g.

```suggestion
        // UserEntrys with the same email but different properties will throw an error, unless they are all in a deleted state.
```